### PR TITLE
TST: test against CPython 3.13 final

### LINKED ISSUE #16595: CPython 3.13 support (tracker issue)
This issue is to plan and keep track on the effort to add support for CPython 3.13 .
See [PEP 719](https://peps.python.org/pep-0719/) for the release schedule.
At the time of opening, version `3.13.0b2` is available, and NumPy is already publishing nightly wheels for it.

Here's the chronological task list to complete before we can declare support for it.
- [x] pass all tests (#16659)
- [x] #16600 
- [x] #16596 
- [x] publish `cp313` wheels to PyPI (done with astropy 6.1.3)
  - [x] CPython 3.13.0rc1 needs to be released https://pythoninsider.blogspot.com/2024/08/python-3130-release-candidate-1-released.html?m=1
  - [x] update manylinux
    - [x] https://github.com/pypa/manylinux/pull/1652
    - [x] https://github.com/pypa/manylinux/pull/1660
    - [x] https://github.com/pypa/manylinux/actions/runs/10226520340
  - [x] update cibuildwheel
    - [x] https://github.com/pypa/cibuildwheel/pull/1949
    - [x] https://github.com/pypa/cibuildwheel/pull/1950
    - [x] https://github.com/pypa/cibuildwheel/releases/tag/v2.20.0
  - [x] https://github.com/OpenAstronomy/github-actions-workflows/pull/220
  - [x] cut a release or add cp313 wheels to the current latest release (the latter is more cumbersome, see https://github.com/pypa/cibuildwheel/discussions/888)
- [x] https://github.com/astropy/astropy/pull/16848
- [x] #16858 
- [ ] If we want Python 3.13 support in v6.1.x series, we also need to apply #16858 and relevant compatibility fixes to v6.1.x branch.

I'll also list which of our core dependencies already have compatible and stable wheels on PyPI, but note that it isn't a necessary condition to make our own.
 - [x] pyerfa (abi3)
 - [x] numpy v2.1.0 (2024-08-18)
 - [x] PyYAML https://github.com/yaml/pyyaml/issues/818

Supporting the experimental free-threading (a.k.a "nogil") build is a non-goal for this issue (although I note that it's currently blocked by https://github.com/liberfa/pyerfa/issues/152)
Relevant PRs are to be linked in this message.

### LINKED ISSUE #16858: TST: Update test matrix when Python 3.13 stable is out
1. Revert https://github.com/astropy/astropy/pull/16600 because it uses "python dev".
2. Modify regular CI matrix to include one job with Python 3.13 (devdeps?).
3. Maybe also modify one of the pre-PR 16600 cron jobs to use Python 3.13 also.

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
Close #16858 and close #16595
Following #16595, this should be backported

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: .github/workflows/ci_cron_weekly.yml [hunk 1]
```diff
@@ -55,19 +55,14 @@ jobs:
...
+            python: '3.13'
             toxenv: py312-test-devinfra
 
-          - name: Python 3.13-dev with dev versions of key dependencies
-            os: ubuntu-latest
-            python: '3.13-dev'
-            toxenv: py313-test-devdeps
-
-          - name: Python 3.12 on macOS (x86_64) with all optional dependencies
+          - name: Python 3.13 on macOS (x86_64) with all optional dependencies
```
Comment (pllim) [inline .github/workflows/ci_cron_weekly.yml:63]: Python 3.13 with all deps so early in release might not fly...

Comment (neutrinoceros) [reply .github/workflows/ci_cron_weekly.yml:63]: yup. Just trying out my luck here. Will revert if (when ?) it comes down crashing into flames.

File: .github/workflows/ci_cron_weekly.yml [hunk 2]
```diff
@@ -55,19 +55,14 @@ jobs:
...
             os: macos-13
-            python: '3.12'
+            python: '3.13'
             toxenv: py312-test-alldeps
```
Comment (pllim) [inline .github/workflows/ci_cron_weekly.yml:66]: Was wondering why it didn't fail. 😆 
```suggestion
            toxenv: py313-test-alldeps
```

Comment (neutrinoceros) [reply .github/workflows/ci_cron_weekly.yml]: ah, too good to be true I guess. Thanks for catching it ! 😄

File: .github/workflows/ci_cron_weekly.yml (hunk 1 above)
Comment (neutrinoceros) [reply .github/workflows/ci_cron_weekly.yml]: nevermind I just did it in https://github.com/astropy/astropy/issues/17140

**Review by pllim — APPROVED** [review]
Thanks! Let's get this in. The exotic arch failures are unrelated. I think those are from plethora of recent refactoring in units. I will open new issue for that one.

**Comment by pllim** [comment]
@neutrinoceros are you interested in manual backport? If not, I can look into that tomorrow. Thanks!

**Comment by neutrinoceros** [comment]
done in https://github.com/astropy/astropy/pull/17142

**Pull request merged**